homeassistant/components/hunterdouglas_powerview/button.py

```diff
@@ -0,0 +1,124 @@
+"""Buttons for Hunter Douglas Powerview advanced features."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, Final
+
+from aiopvapi.resources.shade import BaseShade, factory as PvShade
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    COORDINATOR,
+    DEVICE_INFO,
+    DOMAIN,
+    PV_API,
+    PV_ROOM_DATA,
+    PV_SHADE_DATA,
+    ROOM_ID_IN_SHADE,
+    ROOM_NAME_UNICODE,
+)
+from .coordinator import PowerviewShadeUpdateCoordinator
+from .entity import ShadeEntity
+
+
+@dataclass
+class PowerviewButtonDescriptionMixin:
+    """Mixin to describe a Button entity."""
+
+    press_action: Callable[[BaseShade], Any]
+
+
+@dataclass
+class PowerviewButtonDescription(
+    ButtonEntityDescription, PowerviewButtonDescriptionMixin
+):
+    """Class to describe a Button entity."""
+
+
+BUTTONS: Final = [
+    PowerviewButtonDescription(
+        key="calibrate",
+        name="Calibrate",
+        icon="mdi:swap-vertical-circle-outline",
+        device_class=ButtonDeviceClass.UPDATE,
```

Comment:
the only other available option is RESTART which doesnt really apply here either - I know it would be an archtectural decision but perhaps another device class enum could be created?

UPDATE was closer to the function of these as it polls the shade directly

Reply:
Oh ok, happy to remove just thought the class was required

Reply:
![image](https://user-images.githubusercontent.com/195327/172856890-0ebb2585-ee30-4a5f-9c81-2981af4f24e5.png)

This device is incorrect in this case, the way it is used in this integration does not match its purpose.